### PR TITLE
Implement dropdown autocomplete

### LINKED ISSUE #1247: Dropdown: in NOT NATIVE MODE: once open, autocomplete should work same as in native mode
In NOT NATIVE MODE: once open, autocomplete should work same as in native mode.

Use case: you have a list with all countries and want to look for italy. Under native dropdown you can open it and start typing It.. and it scrolls down to where Italy is.

### PR DESCRIPTION
**Pull request by markus-walther**
Fixes #1247.

**Replaces stale PR https://github.com/axa-ch/patterns-library/pull/1381**

Since arrow-key navigation has been implemented independently of the stale PR in the meantime, a commit to that effect has been dropped in the new PR.

# Done is when (DoD):
- [x] Modifications within components are reflected by tests.
- [x] A self-review of the code has been performed.
- [x] The modified component properties have been added to the typescript typings.
- [x] Potential design changes have been reviewed by a designer.
- [x] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [x] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [x] The modifications are shortly added to CHANGELOG.md with the issue number.
- [x] The modifications still work in IE.
- [x] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

### CODE REVIEW

**Comment by 01231** [comment]
Please fill out DOD before requesting a review😉

File: src/components/20-molecules/dropdown/index.js [hunk 1]
```diff
@@ -232,14 +237,60 @@ class AXADropdown extends NoShadowDOM {
...
+          _autosuggestDictionary.slice(matchIndexPos),
+          10
+        );
+        // ... and its value
+        const { value } = items[firstSelectedItem];
+        // re-render to visually select matched item
+        this.updateItems(value, 'scrollIntoView');
+      }
+      // next characters start a *new* autosuggestion
+      this._typedSoFar = '';
```
Comment (01231) [inline src/components/20-molecules/dropdown/index.js:300]: nice

**Review by 01231 — COMMENTED** [review]
lgtm but can't test it, because feature-branch-deployment seems to be broker for this pr

File: src/components/20-molecules/dropdown/index.js [hunk 2]
```diff
@@ -439,18 +507,46 @@ class AXADropdown extends NoShadowDOM {
...
+    this._autosuggestDictionary =
+      items
+        .filter(item => !item.disabled)
+        .map((item, index) => {
+          // normalize word for purposes of matching
+          const word = item.name.toLowerCase();
+          // construct a set of all letters in word
+          // (used for an early filter over keystrokes)
+          word.split('').forEach(letter => {
+            letters.add(letter.toLowerCase());
```
Comment (01231) [inline src/components/20-molecules/dropdown/index.js:564]: isn't everything already lowercase?

**Comment by 01231** [comment]
I tested it locally. Works good. Works also in react and on IE.

File: src/components/20-molecules/dropdown/index.js (hunk 2 above)
Comment (markus-walther) [reply src/components/20-molecules/dropdown/index.js:564]: No, it is not. `item.name` can be an arbitrary string, so I lowercase it here and then also lowercase the user input during autosuggest, so they can match up regardless of case differences.

**Comment by raphaellueckl** [comment]
What I noticed during testing:

A native dropdown can be opened, if selected, by using `spacebar`.
A native dropdown option can be selected by hitting `spacebar`.

These could also be implemented in the future if needed, otherwise looks nice functionality wise.

File: src/components/20-molecules/dropdown/index.js [hunk 3]
```diff
@@ -251,14 +256,62 @@ class AXADropdown extends NoShadowDOM {
...
+    } else {
+      // selection of first matching dropdown item by quickly typing a prefix
+      // of the item's 'name' value, e.g. 'afgh' for selecting Afghanistan in a country list
+      this.handleAutosuggest(key);
+    }
+  }
+
+  handleAutosuggest(key) {
+    // ready for autosuggestions, by having focus on the non-native UI?
+    if (document.activeElement !== this.button) {
```
Comment (raphaellueckl) [inline src/components/20-molecules/dropdown/index.js:268]: Is this just a safety measure, or are there actual cases where this is not true, when this method is called?

File: src/components/20-molecules/dropdown/index.js [hunk 4]
```diff
@@ -251,14 +256,62 @@ class AXADropdown extends NoShadowDOM {
...
     e.preventDefault();
     e.stopPropagation();
     // toggle dropdown
-    this.openDropdown(!this.open);
+    const { open } = this;
+    this.openDropdown(!open);
+    if (open) {
+      // note: without setTimeout, scrolling is off by several items, s.t. the selected
+      // element is not visible (at least with maxHeight being set)
+      setTimeout(() => this.scrollIntoView(), 1);
```
Comment (raphaellueckl) [inline src/components/20-molecules/dropdown/index.js:313]: Why 1 in the end and not leaving it out?

File: src/components/20-molecules/dropdown/index.js (hunk 3 above)
Comment (markus-walther) [reply src/components/20-molecules/dropdown/index.js:268]: Safety measure, defensive programming

File: src/components/20-molecules/dropdown/index.js (hunk 4 above)
Comment (markus-walther) [reply src/components/20-molecules/dropdown/index.js:313]: uh, timeout needs a ... time argument.

File: src/utils/debounce.js [hunk 5]
```diff
@@ -18,7 +18,7 @@ const debounce = function debounce(
    * @function
    */
   function debounced(...rest) {
-    // cash stuff in closure
+    // cache stuff in closure
```
Comment (raphaellueckl) [inline src/utils/debounce.js:21]: haha :D nice correction

**Review by raphaellueckl — APPROVED** [review]
lgtm

**Pull request merged**